clang/test/CodeGen/fake-use.cpp

```diff
@@ -0,0 +1,44 @@
+// RUN: %clang_cc1 %s -O2 -emit-llvm -fextend-this-ptr -o - | FileCheck %s
+// Check that we generate a fake_use call with the 'this' pointer as argument.
+// The call should appear after the call to bar().
```

Comment:
Hard to see why the rest of the class information in this test is necessary? There's a whole ton of fake uses generated that aren't tested for.

IMO: this is a good test for some other things, like which of the variables get fake uses (not params but yes locals?) and how the lifetime of the locally-allocated object 'V' is handled: seemingly not because it's returned, but what would happen if it wasn't?